src/app/content/components/Page/PageComponent.tsx

```diff
@@ -47,6 +47,19 @@ export default class PageComponent extends Component<PagePropTypes> {
 
       transformContent(parsedContent, parsedContent.body, this.props.intl);
 
+      if (this.props.lockNavigation) {
+        const links = parsedContent.body.querySelectorAll('a[href]');
+
+        for (const link of links) {
+          const {origin, pathname} = new URL(link.getAttribute('href'), assertWindow().location.href);
+          const loc = assertWindow().location;
+
+          if (origin !== loc.origin || pathname !== loc.pathname) {
```

Comment:
there is logic elsewhere (in recipes) that does (or is supposed to) add target blank to any link outside the content, so you don't have to handle that case here. (i think thats what you're looking at the origin for)